src/cleanup_rules/swift/rules.toml

```
@@ -51,6 +51,49 @@ groups = ["boolean_expression_simplify"]
 replace_node = "conjunction_expression"
 replace = "@lhs"
 
+#
+# Before 
+#   true || abcd()
+# After 
+#   true
+#
+[[rules]]
+name = "true_or_something"
+query = """(
+(disjunction_expression
+        lhs: [(boolean_literal) @true  
+            (tuple_expression 
+                value: (boolean_literal) @true)]
+        rhs: (_) 
+    ) @disjunction_expression
+(#eq? @true "true")
+)"""
+groups = ["boolean_expression_simplify"]
+replace_node = "disjunction_expression"
+replace = "true"
+
+
+#
+# Before 
+#   abcd() || true
+# After 
+#   true
+#
+[[rules]]
+name = "something_or_true"
+query = """(
+(disjunction_expression
+        lhs: (_)
+        rhs: [(boolean_literal) @true  
+            (tuple_expression 
+                value: (boolean_literal) @true)]
+    ) @disjunction_expression
+(#eq? @true "true")
+)"""
+groups = ["boolean_expression_simplify"]
+replace_node = "disjunction_expression"
+replace = "true"
```

```suggestion
replace = "@lhs"
```

Since _something_ can have side effects we dont want to remove it.

---

But if we replace `true || sample_function()` with `sample_function()` and if `sample_function()` returns false, wouldn't that be wrong?

---

Made changes as per our offline discussion.